yt/yql/plugin/bridge/plugin.cpp

```diff
@@ -169,6 +172,46 @@ class TYqlPlugin
         return queryResult;
     }
 
+    TClustersResult GetUsedClusters(
+        TString queryText,
+        NYson::TYsonString settings,
+        std::vector<TQueryFile> files) noexcept override
+    {
+        auto settingsString = settings ? settings.ToString() : "{}";
+
+        std::vector<TBridgeQueryFile> filesData;
+        filesData.reserve(files.size());
+        for (const auto& file : files) {
+            filesData.push_back(TBridgeQueryFile{
+                .Name = file.Name.data(),
+                .NameLength = file.Name.size(),
+                .Content = file.Content.data(),
+                .ContentLength = file.Content.size(),
+                .Type = file.Type,
+            });
+        }
+
+        auto* bridgeClustersResult = BridgeGetUsedClusters(
+            BridgePlugin_,
+            queryText.data(),
+            settingsString.data(),
+            settingsString.length(),
```

Comment:
Why do we need pass `settings` string length here? Query text doesn't require that.

Reply:
The value 0 of one of the enum settings fields here can be serialized as '\0'. In order not to lose half of the string, we remember its length.

https://github.com/ytsaurus/ytsaurus/blob/main/yt/yt/server/query_tracker/yql_engine.cpp#L52

It was probably possible to just use the settings conversion into EYsonFormat::Text instead of EYsonFormat::Binary in all places. But historically it was done this way

In the case of the query text, we assume that the correct query would not have any '\0' character, so we don't worry about its length